TM1py/Services/HierarchyService.py

```diff
@@ -356,9 +356,10 @@ def remove_edges_under_consolidation(self, dimension_name: str, hierarchy_name:
         elements_under_consolidations = element_service.get_members_under_consolidation(dimension_name, hierarchy_name,
                                                                                         consolidation_element)
         elements_under_consolidations.append(consolidation_element)
+        elements_under_consolidations = [member.lower().replace(' ','') for member in elements_under_consolidations]
```

Comment:
Have look at *Utils.py*. It offers a lot of helper functions. So  you can `use lower_and_drop_spaces()` for that. Using the helper function improves readability and reduces duplicated code. 
[Link](
https://github.com/cubewise-code/tm1py/blob/2aacccef23737bd9849c6c17f85fa5b10902f4ff/TM1py/Utils/Utils.py#L916)

It would be better to have a separate commit for the bugfix, which only includes the changes for the bugfix. The benefits are:
* It’s easier to understand the commit history
* If there is case where the feature commits must be reverted, the fix remains. 
* The maintainer of the code can cherry pick the bugfix to deliver the bugfix earlier or when your feature request would be declined. 

Maybe it might also be worth to have a own branch/pull request to separate the bugfix from the new feature and allows independent merging.

Reply:
Thank you @onefloid, I will explore the function you suggested here for the purpose. Will try isolating the merge request as well.